Exploration API (#233)
qdrant/landing_page: qdrant-landing/content/articles/exploration-api.md

{{< figure width=80% src=/articles_data/exploration-api/feature-based-recommendations.png caption="Vector-Features Based Recommendations" >}}

This approach is already implemented in Qdrant, and while it works great when the vectors are assumed to have each of their dimensions represent some kind of feature of the data, sometimes distances are a better tool to judge negative and positive examples.

### Relative Distance Recommendations

Another approach is to use the distance between negative examples to the candidates to help them create exclusion areas.
In this technique, we perform searches near the positive examples while excluding the points that are closer to a negative example than to a positive one.

{{< figure width=80% src=/articles_data/exploration-api/relative-distance-recommendations.png caption="Relative Distance Recommendations" >}}

The main use-case of both approaches —of course— is to take some history of user interactions and recommend new items based on it.

## Discovery

In many exploration scenarios, the desired destination is not known in advance.

> Search process in this case can consist of multiple steps, where each step would provide a little more information to guide the search in the right direction.

Reviewer: ```suggestion
The search process in this case can consist of multiple steps, where each step would provide a little more information to guide the search in the right direction.
```

To get more intuition about the possible ways to implement this approach, let’s take a look at how similarity modes are trained in the first place:

The most well-known loss function used to train similarity models is a [triplet-loss](https://en.wikipedia.org/wiki/Triplet_loss).
In this loss, the model is trained by fitting the information of relative similarity of 3 objects: the Anchor, Positive, and Negative examples.

{{< figure width=80% src=/articles_data/exploration-api/triplet-loss.png caption="Triplet Loss" >}}

Using the same mechanics, we can look at the training process from the other side.

> We already have a trained model. The user can provide positive and negative examples, and the goal of the discovery process - is to find suitable anchors across the stored collection of vectors.

Reviewer: ```suggestion
Given a trained model, the user can provide positive and negative examples, and the goal of the discovery process is then to find suitable anchors across the stored collection of vectors.
```
